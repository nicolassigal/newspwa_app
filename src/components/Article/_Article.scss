.article {
    background: var(--bg-card-color);
    border: 1px solid rgba(142, 142, 142, 0.1);
    padding: 20px 10px;
    box-shadow: var(--shadow);
    text-decoration: none;
    color: var(--txt-color);
    &:active,
    &:focus,
    &:active:focus {
        text-decoration: none;
        color: var(--txt-color);
        
    }

    &:hover {
        text-decoration: none;
        color: var(--txt-color);
        box-shadow: var(--shadow-dark);
        cursor: pointer;
    }

    &__title {
        font-size: 16px;
        font-weight: bold;
        line-height: 16px;
        padding-bottom: 10px;
    }
    &__description {
        font-size: 14px;
        font-weight: light;
        line-height: 14px;
    }
    &__picture {
        width: 100%;
        height: 180px;
        object-fit: contain;
        &.preload{
            opacity: 0;
            background: var(--preload);
            min-width: 100%;
            filter: blur(3px);
            min-height: 180px;
            animation: flickerAnimation 3s infinite;
        }
    }
}

@keyframes flickerAnimation {
    0%   { opacity:0; }
    50%  { opacity:1; }
    100% { opacity:0; }
  }