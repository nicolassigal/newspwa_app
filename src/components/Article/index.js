import React, {PureComponent} from 'react';
import Image from '../Image';
class Article extends PureComponent {
    constructor(props) {
        super(props);
        this.state = {loading: true}
        this.loadHandler = this.loadHandler.bind(this);
    }

    loadHandler() {
        this.setState({loading: false})
    }

    render() {
        const {loading} = this.state;
        return (
            <a href={this.props.url}  className="article">
                <article>
                    <Image
                        className={`article__picture ${loading ? 'preload' : ''}`}
                        onLoad={this.loadHandler}
                        src={this.props.image} />
                    <h1 className="article__title">{this.props.title}</h1>
                    <p className="article__description">{this.props.description}</p>
                </article>
            </a>
        )
    }
}

export default Article;