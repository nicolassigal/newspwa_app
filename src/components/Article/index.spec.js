import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Article from './index';
import { shallow } from 'enzyme';

describe('Article component test', () => {
  it('Article component should render correctly', () => {
    const props = {
      title: "title",
      description: "desc",
      image: "img",
      url:"url"
    }


   const wrapper = shallow(<Article {...props}/>)
   expect(wrapper).toMatchSnapshot();

  });
});