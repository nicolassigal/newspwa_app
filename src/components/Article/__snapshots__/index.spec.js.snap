// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article component test Article component should render correctly 1`] = `ShallowWrapper {}`;
